### PR TITLE
#368  Adding support for listing all invoice line items

### PR DESCRIPTION
**Pull request by daniel-cohen**
See: https://stripe.com/docs/api/curl#invoice_lines

### CODE REVIEW

**Comment by jaymedavis** [comment]
looks like async still missing? You can just push to your fork - you don't have to close and reopen. I can merge on my end

**Comment by daniel-cohen** [comment]
@jaymedavis. Check it out. I don't have time to test it, But I think that's what you're after.

File: src/Stripe/Services/Invoices/StripeInvoiceService.cs [hunk 1]
```diff
@@ -53,6 +53,14 @@ public virtual IEnumerable<StripeInvoice> List(StripeInvoiceListOptions listOpti
             );
         }
 
+        public virtual IEnumerable<StripeInvoiceLineItem> ListLineItems(string invoiceId, StripeInvoiceListLineItemsOptions listOptions = null, StripeRequestOptions requestOptions = null)
```
Comment (jaymedavis) [inline src/Stripe/Services/Invoices/StripeInvoiceService.cs]: this should just be called List, to match the rest of the library. And these two methods should be moved to a new InvoiceLineItemService - this follows their documentation exactly. If it doesn't - trust me - I get many emails. ;)

File: src/Stripe/Services/Invoices/StripeInvoiceService.cs [hunk 2]
```diff
@@ -106,6 +114,14 @@ public virtual StripeInvoice Create(string customerId, StripeInvoiceCreateOption
             );
         }
 
+        public virtual async Task<IEnumerable<StripeInvoiceLineItem>> ListLineItemsAsync(string invoiceId, StripeInvoiceListLineItemsOptions listOptions = null, StripeRequestOptions requestOptions = null)
```
Comment (jaymedavis) [inline src/Stripe/Services/Invoices/StripeInvoiceService.cs]: same here for List

**Comment by jaymedavis** [comment]
Actually, there is an InvoiceItemService that appears to handle all of this... let me know if that doesn't meet your needs or I am missing something.

**Comment by jaymedavis** [comment]
Scratch that, I found it in the docs. Sorry about that. :)

**Comment by daniel-cohen** [comment]
It's not the same. This is to list all the line items on a specific invoice. since it's limited to 10 or 11 by default when you get the invoice itself.

**Comment by jaymedavis** [comment]
Man - I was totally confused on this PR, disregard my comments. This makes sense now - the invoice line items is just confusing to think about, but I see what you are doing. Thanks!

**Comment by jaymedavis** [comment]
This is on the invoice-line-items branch. I only made a couple small tweaks to the test, and added the subscription plan id to the options. Excellent work, will be in the next release.

**Comment by daniel-cohen** [comment]
@jaymedavis Thanks for merging this !!

**Pull request merged**